tests/vm_test.go

```diff
@@ -1328,29 +1322,23 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					// compute container of the virt-launcher pod is kept in the running state.
 					// If the annotation v1.KeepLauncherAfterFailureAnnotation is set to false or not set, the virt-launcher pod will become failed.
 					By("Verify that the virt-launcher pod or its container is in the expected state")
-					vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					launcherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
-					Expect(err).ToNot(HaveOccurred())
-
 					if toKeep, _ := strconv.ParseBool(keepLauncher); toKeep {
-						Consistently(func() bool {
-							for _, status := range launcherPod.Status.ContainerStatuses {
-								if status.Name == "compute" && status.State.Running != nil {
-									return true
-								}
-							}
-							return false
-						}, 10*time.Second, 1*time.Second).Should(BeTrue())
+						Consistently(func() error {
+							_, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
+							return err
+						}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
```

Comment:
We should check the Pod is still running and is not terminating.

Reply:
Yes, this verification is wrong; the pod should exist anyway.

> We should check the Pod is still running and is not terminating.

After digging into it, I think we should check if the compute container is still running. :)